<launch>

<!---

Copyright (c) 2012, Falkor Systems, Inc.  All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are
met:

Redistributions of source code must retain the above copyright notice,
this list of conditions and the following disclaimer.  Redistributions
in binary form must reproduce the above copyright notice, this list of
conditions and the following disclaimer in the documentation and/or
other materials provided with the distribution.  THIS SOFTWARE IS
PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

-->

  <node name="ardrone_driver" pkg="ardrone_autonomy" type="ardrone_driver" >
    <param name="realtime_navdata" value="true" />
  </node>

  <node name="ros_tld_tracker_node" pkg="tld_tracker" type="ros_tld_tracker_node" >
    <remap from="image_rgb" to="/ardrone/front/image_raw"/>
    <remap from="bounding_box" to="tld_gui_bb"/>
    <remap from="cmds" to="tld_gui_cmds"/>
    <param name="showOutput" value="true" type="bool"/>
    <param name="loadModel" value="false" type="bool"/>
    <param name="autoFaceDetection" value="false" type="bool"/>
    <param name="modelImportFile" value="$(find falkor_ardrone)/cascade/falkor_tld_model" type="string"/>
    <param name="exportModelAfterRun" value="true" type="bool"/>
    <param name="modelExportFile" value="$(find falkor_ardrone)/cascade/falkor_tld_model_new" type="string"/>
  </node>

  <node name="ros_tld_gui_node" pkg="tld_gui" type="ros_tld_gui_node" >
    <remap from="image_rgb" to="/ardrone/front/image_raw"/>
    <remap from="tracked_object" to="tld_tracked_object"/>
    <remap from="fps_tracker" to="tld_fps"/>
  </node>

  <node name="ardrone_follow" pkg="falkor_ardrone" type="ardrone_follow.py" output="screen" />

  <node name="ardrone_control" pkg="falkor_ardrone" type="ardrone_control.py" />

  <node name="joy_node" pkg="joy" type="joy_node" >
    <param name="dev" value="/dev/input/js1" />
  </node>

</launch>
